extensions/http/deployment/src/main/java/org/apache/camel/quarkus/component/http/deployment/HttpProcessor.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.http.deployment;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+class HttpProcessor {
+    public static final String FEATURE = "camel-http";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(FEATURE);
```

Comment:
I think it is the component that has the option to use ssl or not

Reply:
> Would not HttpClientProcessor be a better place for this?
> 

 think it is the component that has the option to use ssl or not so I think the suppor extension should not impose to bring ssl into the game

> BTW, if we auto-enable SSL, should we not auto-enable NativeImageEnableAllCharsetsBuildItem too? (I assume the HTTP client offers the option to decode the response body using the encoding (which can be any encoding) sent by the server)

This may be a separate issue as we may have other extensions affected, minding opening an additional issue ?

Reply:
> I think it is the component that has the option to use ssl or not

(Thinking loud, not imposing any particular view) That would certainly be OK, if we were likely to have components using http client that would not need HTTPS. Are we really likely to have HTTP-client-dependent components not needing HTTPS?

Asking also because the AHC has it in support module and it would perhaps be practical to do it similarly for both. (I am not implying which of the two needs to be fixed)

Reply:
probably not but I think support extension should in general not force specific behaviours.
I think one of the goal of such support extension is to be moved at quarkus soon or later so I'd go for do the bare minimum.

Reply:
Created https://github.com/apache/camel-quarkus/issues/599